src/unread/__tests__/unreadModel-test.js

```diff
@@ -0,0 +1,188 @@
+            streams: [
+              {
+                stream_id: message1.stream_id,
+                topic: message1.subject,
+                unread_message_ids: [message1.id, 2],
+              },
+            ],
+            huddles: [],
+            pms: [],
+            mentions: [message1.id, 2, 3],
+          },
+        },
+      };
+
+      // prettier-ignore
+      expect(summary(reducer(initialState, action))).toEqual(Immutable.Map([
+        [message1.stream_id, Immutable.Map([[message1.subject, [1, 2]]])],
+      ]));
+    });
+  });
+
+  describe('EVENT_NEW_MESSAGE', () => {
+    const action = mkMessageAction;
+
+    const baseState = (() => {
+      let state = initialState;
+      state = reducer(state, action(eg.streamMessage({ id: 1, subject: 'some topic' })));
+      return state;
+    })();
+
+    test('(base state, for comparison)', () => {
+      // prettier-ignore
+      expect(summary(baseState)).toEqual(Immutable.Map([
+        [eg.stream.stream_id, Immutable.Map([['some topic', [1]]])],
+      ]));
+    });
+
+    test('if message id already exists, do not mutate state', () => {
+      const state = reducer(baseState, action(eg.streamMessage({ id: 1, subject: 'some topic' })));
+      expect(state).toBe(baseState);
+    });
+
+    test('if message is not stream, return original state', () => {
+      const state = reducer(baseState, action(eg.pmMessage({ id: 4 })));
+      expect(state.streams).toBe(baseState.streams);
+    });
+
+    test('if message is sent by self, do not mutate state', () => {
+      const state = reducer(baseState, action(eg.streamMessage({ sender: eg.selfUser })));
+      expect(state).toBe(baseState);
+    });
+
+    test('if message id does not exist, append to state', () => {
+      const state = reducer(baseState, action(eg.streamMessage({ id: 4, subject: 'some topic' })));
+      // prettier-ignore
+      expect(summary(state)).toEqual(Immutable.Map([
+        [eg.stream.stream_id, Immutable.Map([['some topic', [1, 4]]])],
+      ]));
+    });
+
+    test('if stream with topic does not exist, append to state', () => {
+      const message = eg.streamMessage({ id: 4, stream_id: 2, subject: 'another topic' });
+      const state = reducer(baseState, action(message));
+      // prettier-ignore
+      expect(summary(state)).toEqual(Immutable.Map([
+        [eg.stream.stream_id, Immutable.Map([['some topic', [1]]])],
+        [message.stream_id, Immutable.Map([['another topic', [4]]])],
+      ]));
+    });
+  });
+
+  describe('EVENT_UPDATE_MESSAGE_FLAGS', () => {
+    const mkAction = args => {
+      const { all = false, messages, flag = 'read', operation = 'add' } = args;
+      return {
+        id: 1,
+        type: EVENT_UPDATE_MESSAGE_FLAGS,
+        allMessages: {},
+        all,
+        messages,
+        flag,
+        operation,
+      };
+    };
+
+    const baseState = (() => {
+      const streamAction = args => mkMessageAction(eg.streamMessage(args));
```

Comment:
It could, but I'm inclined to keep the name short: this helper is local to a quite short stretch of code, so it's always easy to go look at its definition. And I like that the `state = reducer(state, …);` statements fit on one line each, and those lines are already kind of long.

If this were a more global name (like if it were exported, or even exposed to this whole test file), then I'd definitely want to give it a more explicit name like `streamMessageAction`.